python/cuspatial/cuspatial/core/binpreds/binpred_interface.py

```diff
@@ -0,0 +1,559 @@
+    """Result of a binary predicate operation."""
+
+    pass
+
+
+class ContainsOpResult(OpResult):
+    """Result of a binary predicate operation.
+
+    Parameters
+    ----------
+    result : cudf.DataFrame
+        A cudf.DataFrame containing two columns: "polygon_index" and
+        Point_index". The "polygon_index" column contains the index of
+        the polygon that contains each point. The "point_index" column
+        contains the index of each point that is contained by a polygon.
+    points : GeoSeries
+        A GeoSeries of points.
+    point_indices : cudf.Series
+        A cudf.Series of indices that map each point in `points` to its
+        corresponding feature in the right-hand GeoSeries.
+
+    Attributes
+    ----------
+    result : cudf.DataFrame
+        A cudf.DataFrame containing two columns: "polygon_index" and
+        Point_index". The "polygon_index" column contains the index of
+        the polygon that contains each point. The "point_index" column
+        contains the index of each point that is contained by a polygon.
+    points : GeoSeries
+        A GeoSeries of points.
+    point_indices : cudf.Series
+        A cudf.Series of indices that map each point in `points` to its
+        corresponding feature in the right-hand GeoSeries.
+    """
+
+    def __init__(
+        self,
+        result: Series,
+        points: "GeoSeries" = None,
+        point_indices: Series = None,
+    ):
+        self.result = result
+        self.points = points
+        self.point_indices = point_indices
+
+    def __repr__(self):
+        return f"OpResult(result={self.result}, points={self.points}, \
+        point_indices={self.point_indices})"
+
+    def __str__(self):
+        return self.__repr__()
+
+
+class EqualsOpResult(OpResult):
+    """Result of a binary predicate operation.
+
+    Parameters
+    ----------
+    result : cudf.Series
+        A cudf.Series of boolean values indicating whether each feature in
+        the right-hand GeoSeries is equal to the point in the left-hand
+        GeoSeries.
+    point_indices: cudf.Series
+        A cudf.Series of indices that map each point in `points` to its
+        corresponding feature in the right-hand GeoSeries.
+
+    Attributes
+    ----------
+    result : cudf.Series
+        A cudf.Series of boolean values indicating whether each feature in
+        the right-hand GeoSeries is equal to the point in the left-hand
+        GeoSeries.
+    point_indices: cudf.Series
+        A cudf.Series of indices that map each point in `points` to its
+        corresponding feature in the right-hand GeoSeries.
+    """
+
+    def __init__(self, result: Series, point_indices: Series):
+        self.result = result
+        self.point_indices = point_indices
+
+    def __repr__(self):
+        return f"OpResult(result={self.result}) \
+        point_indices={self.point_indices}"
+
+    def __str__(self):
+        return self.__repr__()
+
+
+class BinPredItf(ABC):
```

Comment:
What does `Itf` stand for?

Reply:
Interface. It is a PIMPL notation, in the past I've worked on multiple C++ projects that used PIMPL and `Itf` for the abstract class and `Impl` for the concrete.

Reply:
Resolved by dropping `BinPredItf`.